src/fluree/db/branch.cljc

```diff
@@ -59,7 +59,9 @@
   [{:keys [alias commit t] :as current-db} index-map]
   (if (async-db/db? current-db)
     (dbproto/-index-update current-db index-map)
-    (let [updated-commit (assoc commit :index index-map)
+    (let [updated-commit (-> commit
+                             (assoc :index index-map)
+                             (assoc :alias alias))  ; Ensure alias is on the commit
           updated-db     (async-db/->async-db alias updated-commit t)]
       (go ;; update index in the background, return updated db immediately
         (let [db* (<? (dbproto/-index-update current-db index-map))]
```

Comment:
This looks like it still expects `-index-update` to return a channel. It [doesn't return a channel for FlakeDB](https://github.com/fluree/db/blob/fa442c4c67171359e1459a2304f2f3b5044c261b/src/fluree/db/flake/flake_db.cljc#L99-L115), and the changes here mean that it doesn't for AsyncDB either.

Reply:
Line 65 is still calling `<?` on the result of `dbproto/-index-update`, but that result will never be a channel. This will result in an exception from trying to take on something that isn't a channel that will then be silently swallowed by the wrapping `go` block.